python/tests/test_lowlevel.py

```diff
@@ -489,6 +497,30 @@ def test_direct_instantiation(self):
         assert "IBD" in output
 
 
+class TestIbdSegmentList:
+    def test_direct_instantiation(self):
+        # Nobody should do this, but just in case
+        seglist = _tskit.IbdSegmentList()
+        attrs = ["num_segments", "total_span", "left", "right", "node"]
+        for attr in attrs:
+            with pytest.raises(SystemError, match="not initialised"):
+                getattr(seglist, attr)
+
+    def test_memory_management(self):
+        ts = msprime.simulate(10, random_seed=1)
+        tc = ts.tables._ll_tables
+        result = tc.ibd_segments()
+        del ts, tc
+        lst = result.get(0, 1)
+        assert lst.num_segments == 1
+        del result
+        assert lst.num_segments == 1
```

Comment:
Is it worth adding some `gc.collect()` after the `del` to be ultra paranoid here?

Reply:
Sure! Good call.